client/app/components/composites/Menu/Menu.js

```diff
@@ -40,9 +47,24 @@ class Menu extends Component {
     });
   }
 
+  handleMouseover() {
+    clearTimeout(this.mouseOutTimout);
+    clearTimeout(this.mouseOverTimout);
+    this.mouseOverTimout = setTimeout(() => (
+      this.setState({ isHovering: true, isOpen: true })  // eslint-disable-line react/no-set-state
+      ), HOVER_TIMEOUT);
+  }
+
+  handleMouseout() {
+    clearTimeout(this.mouseOverTimout);
+    this.mouseOutTimout = setTimeout(() => (
+      this.setState({ isHovering: false, isOpen: false }) // eslint-disable-line react/no-set-state
+      ), HOVER_TIMEOUT);
+  }
+
```

Comment:
Maybe no need to change at this point, but this uses the same hovering/timeout code that the other component above and could be abstracted into a HOC that wraps these components and provides a `hover` and `open` flags etc.